ibis/common/selectors.py

```diff
@@ -29,3 +28,79 @@ def expand(self, table: ir.Table) -> Sequence[ir.Value]:
             A sequence of value expressions that match the selector
 
         """
+
+    @abc.abstractmethod
+    def expand_names(self, table: ir.Table) -> frozenset[str]:
```

Comment:
Using a `frozenset` here feels a bit weird to me. I understand that sets are now technically ordered unique collections, but I guess after years of python usage where that wasn't true relying on the ordering properties of a `frozenset` (as opposed to a `tuple`/`list`) in an interface feels wrong. Like, if I got back a `frozenset` from an API, I wouldn't intuitively think that the ordering of that set was meaningful or mattered. Would you be open to changing this to a `tuple[str]` or `list[str]` instead? The uniqueness properties could still be enforced by the implementations without requiring it in the return type.

Reply:
Thinking further, the generation here of names uses a lot of the `frozenset` apis for ease, perhaps using it as a return type is fine. It's not user facing really anyway. :shrug:.

Reply:
The `frozenset` here is also used for fast membership testing in a number of places, it's not only about uniqueness.